### PR TITLE
Add events to notify consumers when messages are acked & nacked

### PR DESCRIPTION
**Pull request by enigmatic00**

### CODE REVIEW

**Comment by binaryalchemist** [comment]
Please bump the version number in the package.json to 0.4.0.

File: src/busterbunny.js [hunk 1]
```diff
@@ -27,7 +27,9 @@ module.exports = (function() {
             AMQP_ERROR: 'amqp-error',
             PUBLISH_CHANNEL_ESTABLISHED: 'publish-channel-established',
             PUBLISH_REQUESTED: 'publish-requested',
-            EVENT_RECEIVED: 'event-received'
+            EVENT_RECEIVED: 'event-received',
```
Comment (binaryalchemist) [inline src/busterbunny.js:38]: These events should be added to the Readme.

File: src/busterbunny.js [hunk 2]
```diff
@@ -169,15 +171,19 @@ module.exports = (function() {
                             var messageObj = {
                                 reject: function (requeue) {
                                     channel.nack(message, false, requeue);
+                                    self.emit(self.EVENTS.EVENT_NACKED, message, requeue, new Date().getTime());
```
Comment (binaryalchemist) [inline src/busterbunny.js:187]: Nacking should update something in the stats object called "messagesRejected".

File: src/busterbunny.js [hunk 3]
```diff
@@ -169,15 +171,19 @@ module.exports = (function() {
...
+
                                 },
                                 acknowledge: function () {
                                     channel.ack(message, false)
+                                    self.emit(self.EVENTS.EVENT_ACKED, message, new Date().getTime());
```
Comment (binaryalchemist) [inline src/busterbunny.js:193]: Acking should update something in the status object called "messagesAcknowledged".

File: src/busterbunny.js [hunk 4]
```diff
@@ -169,15 +171,19 @@ module.exports = (function() {
...
                                 }
                             };
 
                             self.emit(self.EVENTS.EVENT_RECEIVED, event, messageObj);
                         } catch (err) {
                             channel.nack(message, false, true);
+                            self.emit(self.EVENTS.EVENT_NACKED, message, false, new Date().getTime());
```
Comment (binaryalchemist) [inline src/busterbunny.js:201]: Nacking should update something in the stats object called "messagesRejected".

File: src/busterbunny.js (hunk 1 above)
Comment (enigmatic00) [inline src/busterbunny.js:38]: Added

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "busterbunny",
-  "version": "0.3.0",
+  "version": "0.4.0",
```
Comment (enigmatic00) [inline package.json:3]: Version bumped

File: src/busterbunny.js [hunk 6]
```diff
@@ -169,15 +179,26 @@ module.exports = (function() {
                             var messageObj = {
                                 reject: function (requeue) {
                                     channel.nack(message, false, requeue);
+                                    if(requeue && requeue === true){
```
Comment (radavi) [inline src/busterbunny.js]: Shouldn't this be `if(requque === true) {`?

**Pull request merged**